frontend/src/presentation/pages/WorkspacePage.tsx

```diff
@@ -0,0 +1,187 @@
+					setSelectedQuestion(null);
+					setIsWorking(false);
+				}
+			} else {
+				setSelectedQuestion(null);
+				setIsWorking(false);
+			}
+		};
+
+		window.addEventListener("hashchange", handleHashChange);
+		handleHashChange();
+		return () => window.removeEventListener("hashchange", handleHashChange);
+	}, []);
+
+	const handleSelectQuestion = useCallback(async (questionId: string) => {
+		setIsLoading(true);
+		setError(null);
+		try {
+			const question = await questionUseCases.getQuestion(questionId);
+			setSelectedQuestion(question);
+			setIsWorking(false);
+			window.location.hash = `#${question.id}`;
+		} catch (err) {
+			setError(handleError(err, ERRORS.FAILED_TO_LOAD_SELECTED_QUESTION));
+		} finally {
+			setIsLoading(false);
+		}
+	}, []);
+
+	const handleStartQuestion = useCallback(() => {
+		setIsWorking(true);
+		if (selectedQuestion) {
+			window.location.hash = `#${selectedQuestion.id}${HASH.SEPARATOR}${ROUTES.WORKING}`;
+		}
+	}, [selectedQuestion]);
+
+	const handleBreadcrumbClick = useCallback(
+		(item: string) => () => {
+			if (item === ROUTES.WORKSPACE) {
+				setSelectedQuestion(null);
+				setIsWorking(false);
+				window.location.hash = "";
+			} else if (item === selectedQuestion?.title) {
+				setIsWorking(false);
+				window.location.hash = `#${selectedQuestion.id}`;
+			}
+		},
+		[selectedQuestion]
+	);
+
+	const isNarrow = selectedQuestion !== null;
+
+	const renderBreadcrumb = () => (
+		<Breadcrumb className={styles.breadcrumb}>
+			<Breadcrumb.Item>
+				<Button type="link" onClick={handleBreadcrumbClick(ROUTES.WORKSPACE)}>
+					{ROUTES.WORKSPACE}
+				</Button>
+			</Breadcrumb.Item>
+			{selectedQuestion && (
+				<Breadcrumb.Item>
+					<Button
+						type="link"
+						onClick={handleBreadcrumbClick(selectedQuestion.title)}
+					>
+						{selectedQuestion.title}
+					</Button>
+				</Breadcrumb.Item>
+			)}
+			{isWorking && (
+				<Breadcrumb.Item>
+					{ROUTES.WORKING.charAt(0).toUpperCase() + ROUTES.WORKING.slice(1)}
+				</Breadcrumb.Item>
+			)}
+		</Breadcrumb>
+	);
+
+	return (
+		<div className={styles.container}>
+			{renderBreadcrumb()}
+			<h1 className={styles.heading}>{ROUTES.WORKSPACE}</h1>
+			<p className={styles.description}>
+				This is your workspace. Below are the available questions.
+			</p>
+
+			<div className={styles.scrollContainer}>
+				<Row
+					className={`${styles.contentRow} ${
+						selectedQuestion ? styles.withSelection : ""
+					}`}
+					gutter={32}
+				>
+					<Col
+						span={isWorking ? 8 : selectedQuestion ? 8 : 24}
+						className={styles.transitionCol}
+					>
+						{!isWorking ? (
+							<QuestionList
+								isNarrow={isNarrow}
+								questions={questions}
+								selectedQuestion={selectedQuestion}
+								onSelectQuestion={handleSelectQuestion}
+								isLoading={isLoading}
+								error={error}
+							/>
+						) : (
+							selectedQuestion && (
+								<QuestionDetail
```

Comment:
If I am understanding your code correctly, does this mean that if the user is working, the list will no longer be visible? if so, do refer to the UI wireframe, because this is different from what is there. The list should be visible at all times. Only the left half of the screen should change at all. Landing component, details page (appearing as how it would to normal users), and editing/adding page. If you feel that this UI is better, please consult with the team first. (I mean, it should be ok in this case, but if this happens for something important later on, we may have bigger issues).

Reply:
![image](https://github.com/user-attachments/assets/94c05432-32e9-4cf2-b2da-2f1bbc673227)
Refer to the ui in figma the list is not visible when user start doing the question, and from the ux perspective it does, not make sense to have the list when doing the question.
Please note that the figma is only for the reference.

Reply:
Wait so this workspace is for answering the question? Not the question list? That should be in a separate service (the Integration and Collaboration Service in Milestone D5). This service (and milestone) is exclusively for question management, not question answering.

Reply:
The question service does not involve the user solving questions at all. It is exclusively for managing the questions. The questions should only be workable on after a match with another user, so it's locked behind milestone D5.